### ISSUE TITLE
CMake build script

### ISSUE THREAD

**Issue by madebr** [AdrienTD/c47edit#16]
Hey, would you be ok with adding a cmake script to the repo?
The reason I'm asking is it allows for cross-platform support,
and it would also allow easier usage on ci.

I played with [caching vcpkg on GitHub actions yesterday](https://github.com/madebr/c47edit/tree/github-actions),
but got stuck with passing vcpkg flags to the msbuild project.
With cmake and vcpkg, it would be much easier to find, and use the vcpkg dependencies.

I'd do this after #15 lands.
I'll leave it up to you to keep or remove the msbuild project,
but the cmake project should do everything the msbuild project does, and more.

**Comment by AdrienTD** [comment]
Yeah, ideally having CMake could be nice to have, but at the same time it would be a major change.
I would have to change how I open the project on my PC and also update the README etc.

I have already used CMake on some of my projects, and they can be opened directly in Visual Studio very well (maybe some small bugs here and there but hopefully fixed since then). So hopefully that won't make it harder to use compared to just open the VS project file in Visual Studio.

**Comment by madebr** [comment]
CMake can generate Visual Studio projects, so you'd still be able to work in the same environment.
Visual Studio 2019+ can open CMake projects immediately, or you can manually run cmake on cli once and work in VS from then on.
I'd have to boot into Windows to modify the script such that it generates nice folders for sources vs headers. I don't have used [`source_group`](https://cmake.org/cmake/help/latest/command/source_group.html) before.
